infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftProject.java

```diff
@@ -0,0 +1,80 @@
+package org.eclipse.che.workspace.infrastructure.openshift;
+
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.api.model.Project;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+
+import java.util.Optional;
+
+/**
+ * TODO Add javadoc
+ *
+ * @author Sergii Leshchenko
+ */
+public class OpenShiftProject {
+    private final OpenShiftPods     pods;
+    private final OpenShiftServices services;
+    private final OpenShiftRoutes   routes;
+
+    public OpenShiftProject(String name,
+                            OpenShiftClientFactory clientFactory) throws InfrastructureException {
+        this.pods = new OpenShiftPods(name, clientFactory);
+        this.services = new OpenShiftServices(name, clientFactory);
+        this.routes = new OpenShiftRoutes(name, clientFactory);
+
+        try (OpenShiftClient client = clientFactory.create()) {
+            Optional<Project> projectOpt = get(name, client);
+            if (!projectOpt.isPresent()) {
+                create(name, client);
+            }
+        }
+    }
+
+    public OpenShiftPods pods() {
+        return pods;
+    }
+
+    public OpenShiftServices services() {
+        return services;
+    }
+
+    public OpenShiftRoutes routes() {
+        return routes;
+    }
+
```

Comment:
Why not to use getter name pattern (getPods(), getServices(), getRoutes()) ? I suppose it is as good practices for public methods unless you have some special reason?